src/Tags.elm

```diff
@@ -0,0 +1,217 @@
+module Tags exposing (commonTagsRule, expressionTagsRule, ruleConfig)
+
+import Elm.Syntax.Expression exposing (Expression(..), LetDeclaration(..))
+import Elm.Syntax.Node as Node exposing (Node(..))
+import ElmSyntaxHelpers
+import Json.Encode as Encode exposing (Value)
+import Review.ModuleNameLookupTable as LookupTable exposing (ModuleNameLookupTable)
+import Review.Rule as Rule exposing (Rule)
+import RuleConfig exposing (AnalyzerRule(..), RuleConfig)
+import Set exposing (Set)
+
+
+type alias Tag =
+    String
+
+
+type alias ProjectContext =
+    Set Tag
+
+
+type alias ModuleContext =
+    { lookupTable : ModuleNameLookupTable
+    , tags : Set Tag
```

Comment:
Should this be ProjectContext?

Reply:
Mmh that's an interesting question. I would say no, because conceptually I'm not trying to hold the ProjectContext inside of the ModuleContext, the ProjectContext happens to only have tags in it, but it could potentially need more things. 
I think a satisfying solution would be to define `type alias ProjectContext = { tags: Set Tag }`.

Reply:
I didn't want to derail the conv so I said nothing before, but now that's on the table ... I tend to forbid all `type alias` that are trivial renames in my code bases. I always prefer having something like `{ context : Set Tag }` instead of `type alias Context = Set Tag`.

Reply:
What would you do for `type alias Tag = String`?
It really does make the code easier to read for me, and wrapping would be quite noisy. Would you wrap it anyway? Or give up and use `String`?

Reply:
```elm
-- I'd change the following
type alias Tag = String
type alias ProjectContext = Set Tag
type alias ModuleContext = { ..., tags : Set Tag }
commonTags : Set Tag
fromProjectToModule : Rule.ContextCreator ProjectContext ModuleContext
fromModuleToProject : Rule.ContextCreator ModuleContext ProjectContext
dataExtractor : ProjectContext -> Value
matchExpression : Node Expression -> Set Tag

-- Into the following (no Tag alias, no ProjectContext alias)
type alias ModuleContext = { ..., tags : Set String }
commonTags : Set String
fromProjectToModule : Rule.ContextCreator { tags : Set String } ModuleContext
fromModuleToProject : Rule.ContextCreator ModuleContext { tags : Set String }
dataExtractor : { tags : Set String } -> Value
-- also changed the name here
tagMatchExpression : Node Expression -> Set String
```